onnxruntime/core/providers/webgpu/program.h

```diff
@@ -233,13 +233,21 @@ struct ProgramInput {
 };
 
 struct ProgramOutput {
+ private:
+  struct AtomicTag {};
+
+ public:
+  constexpr static const AtomicTag Atomic{};
+
   ProgramOutput(Tensor* tensor);
   ProgramOutput(Tensor* tensor, ProgramTensorMetadataDependency dependency, int component = 1);
+  ProgramOutput(Tensor* tensor, ProgramTensorMetadataDependency dependency, AtomicTag);
```

Comment:
Can we  use `is_atomic` flag as argument with default value false instead of using operator overloading here, using AtomicTag,

Reply:
Using tag type helps to give better overload flexibility and better sementic. As described in the PR description, you can use the following way to specify:
```
program.AddOutput({output_tensor, ProgramTensorMetadataDependency::TypeAndRank, ProgramOutput::Atomic});
```